pkg/embeddedcluster/upgrade.go

```diff
@@ -0,0 +1,154 @@
+package embeddedcluster
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	embeddedclusterv1beta1 "github.com/replicatedhq/embedded-cluster-kinds/apis/v1beta1"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	kotsv1beta1 "github.com/replicatedhq/kotskinds/apis/kots/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+const (
+	clusterUpgradeTimeout = 5 * time.Minute
```

Comment:
I don't know if we want a timeout for cluster upgrades as it's not easy to predict its duration. Can vary significantly from one app to another.

Reply:
something has to cancel this as it happens in the background and can go forever

Reply:
Oh this is actually just for distributing the infra images and upgrading the operator chart, not the entire cluster. So it's probably fine to have a timeout.